test/bdd/did_resolver_sidetree_node_steps.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package bdd
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/DATA-DOG/godog"
+	"github.com/go-openapi/swag"
+
+	"github.com/trustbloc/sidetree-core-go/pkg/document"
+	"github.com/trustbloc/sidetree-core-go/pkg/docutil"
+	"github.com/trustbloc/sidetree-node/models"
+)
+
+const sha2256 = 18
+const didDocNamespace = "did:sidetree:"
+
+// DIDResolverSideTreeNodeSteps
+type DIDResolverSideTreeNodeSteps struct {
+	bddContext       *Context
+	reqEncodedDIDDoc string
+	resp             *httpRespone
+}
+
+// NewDIDResolverSteps
+func NewDIDResolverSideTreeNodeSteps(context *Context) *DIDResolverSideTreeNodeSteps {
+	return &DIDResolverSideTreeNodeSteps{bddContext: context}
+}
+
+func (d *DIDResolverSideTreeNodeSteps) createDIDDocument(sideTreeURL, didDocumentPath string) error {
+	req := newCreateRequest(didDocumentPath)
+	d.reqEncodedDIDDoc = swag.StringValue(req.Payload)
+	var err error
+	d.resp, err = sendRequest(sideTreeURL, req)
+	return err
+}
+
+func (d *DIDResolverSideTreeNodeSteps) checkSuccessResp(msg string) error {
+	if d.resp.errorMsg != "" {
+		return fmt.Errorf("error resp %s", d.resp.errorMsg)
+	}
+
+	if msg == "#didDocumentHash" {
+		documentHash, err := docutil.CalculateID(didDocNamespace, d.reqEncodedDIDDoc, sha2256)
+		if err != nil {
+			return err
+		}
+		msg = strings.Replace(msg, "#didDocumentHash", documentHash, -1)
+	}
+	logger.Infof("check success resp %s contain %s", string(d.resp.payload), msg)
+	if !strings.Contains(string(d.resp.payload), msg) {
+		return fmt.Errorf("success resp %s doesn't contain %s", d.resp.payload, msg)
+	}
+	return nil
+}
+
+func newCreateRequest(didDocumentPath string) *models.Request {
+	operation := models.OperationTypeCreate
+	alg := "ES256K"
+	kid := "#key1"
+	payload := encodeDidDocument(didDocumentPath)
+	signature := "mAJp4ZHwY5UMA05OEKvoZreRo0XrYe77s3RLyGKArG85IoBULs4cLDBtdpOToCtSZhPvCC2xOUXMGyGXDmmEHg"
```

Comment:
any reason why this signature is hard coded?

Reply:
done